mac

```diff
@@ -210,5 +236,16 @@ if [ -f "$HOME/.laptop.local" ]; then
   . "$HOME/.laptop.local"
 fi
 
-SHELL=$( which zsh )
-$SHELL -l
+fancy_echo "Script complete!"
+
+if [[ "${SKIP_ZSH}" == 1 ]]; then
+  fancy_echo "Since zsh was not installed, please add the following to your shell rc:"
```

Comment:
What do you think about just adding the changes to `~/.profile` instead? It should be loaded regardless of shell choice, and is POSIX-compliant:
https://superuser.com/a/789465

Reply:
Hm... I could move it into `.profile`.  Is this script ever run in existing environments (with the changes already in `.zshrc`)? Is that something that needs to be supported?

We'll need to remember a few other things too:
- some people's stuff don't source `.profile` (I make a rule not to, so maybe the warning message isn't a bad idea); and
- additional scripts in the future may misbehave when it's a non-interactive shell (luckily `~/.asdf/asdf.sh` checks out).

Reply:
I don't think that anyone ever re-runs this script, but that does add to the complexity.

I originally thought that `~.profile` would _always_ be evaluated, but if that's not true, we shouldn't rely on it.

Perhaps just echoing out the needed changes is good enough for now, we can always add more automation later.